src/acktest/bootstrapping/__init__.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import abc
+import pickle
+import logging
+
+from pathlib import Path
+from dataclasses import dataclass, fields
+from typing import Iterator
+
+@dataclass
+class ServiceBootstrapResources(abc.ABC):
+    """Represents a list of all bootstrappable resources required for a given
+    service's tests.
+    """
+    def serialize(self, output_path: Path, bootstrap_file_name: str = "bootstrap.pkl"):
```

Comment:
should be a `@classmethod` like `deserialize`, yes?

Reply:
No I wanted this to look like `resources.serialize(...)`. Whereas deserialize is like a factory for these types.

Reply:
sorry, I'm not following you... could you elaborate?

Reply:
In my mind: `Serializable` should represent a class that can be serialized and therefore knows how to serialize itself - i.e. calling `resource.serialize()`.
Changing this to a `@classmethod` would work, but I think `Serializable.serialize(resource)` looks as clean.

Reply:
k, fair enough. :)